src/app/store/selectors/VehicleTestResultModel.selectors.ts

```diff
@@ -1,11 +1,43 @@
 import { createSelector, createFeatureSelector } from '@ngrx/store';
-import {IVehicleTestResultModelState} from '../../store/state/VehicleTestResultModel.state';
+import { IVehicleTestResultModelState } from '@app/store/state/VehicleTestResultModel.state';
+import { TestRecordTestType } from '@app/models/test-record-test-type';
+import { TestType } from '@app/models/test.type';
+import { TestResultModel } from '@app/models/test-result.model';
 
 export const selectFeature = createFeatureSelector<IVehicleTestResultModelState>(
   'vehicleTestResultModel'
 );
 
-export const selectSelectedVehicleTestResultModel = createSelector(
+export const getVehicleTestResultModel = createSelector(
+  selectFeature,
+  (state: IVehicleTestResultModelState) => state.vehicleTestResultModel
+);
+
+export const selectTestTypeById = (id: string) =>
+  createSelector(selectFeature, (state: IVehicleTestResultModelState) => {
+    let testType: TestType;
+    let testRecord: TestResultModel;
+
+    if (state.selectedTestResultModel) {
```

Comment:
Can we simplify it with a filter function maybe?
What about something like that
state.selectedTestResultModel.testTypes.filter(testType => testType.testNumber === id)

Reply:
Refactored using projector :D